app/controllers/competitions_controller.rb

```diff
@@ -272,21 +276,34 @@ def payment_integration_setup
 
   def connect_payment_integration
     competition = competition_from_params
-    payment_integration = params.require(:payment_integration)
+    payment_integration = params.require(:payment_integration).to_sym
 
     raise ActionController::RoutingError.new('Not Found') unless current_user&.can_manage_competition?(competition)
 
-    if payment_integration == 'paypal' && PaypalInterface.paypal_disabled?
+    if payment_integration == :paypal && PaypalInterface.paypal_disabled?
       flash[:error] = 'PayPal is not yet available in production environments'
       return redirect_to competitions_payment_setup_path(competition)
     end
 
-    connector = CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[payment_integration.to_sym].safe_constantize
-    account_reference = connector&.connect_account(params)
+    if CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[payment_integration].nil?
```

Comment:
We need to check for the validity of the integration here, before we try and assign it to a variable. Old-line 284 actually fails if the integration is unrecognized because `safe_constantize` can't be called on `nil`